pkg/kncloudevents/message_sender.go

```diff
@@ -160,6 +160,51 @@ func RetryConfigFromDeliverySpec(spec duckv1.DeliverySpec) (RetryConfig, error)
 	return retryConfig, nil
 }
 
-func checkRetry(_ context.Context, resp *nethttp.Response, err error) (bool, error) {
-	return !(resp != nil && resp.StatusCode < 300), err
+// Simple default implementation
+func RetryIfGreaterThan300(_ context.Context, response *nethttp.Response, err error) (bool, error) {
+	return !(response != nil && (response.StatusCode < 300)), err
```

Comment:
I think might want to change the default... to also retry on `-1` status code. What do you think @slinkydeveloper ?

See: https://github.com/knative/eventing/pull/4890/files#diff-0164cfd8eae490fb398be52f3f4ec3386b73fa31ceba43723a63777ccc103a7eR204-R206 

I find that reasonable - but we can do that in a different PR... since that kinda changes behavioir of our default

Reply:
I think it makes sense to change it in a different PR